### ISSUE TITLE
Consider ignoring imports under TYPE_CHECKING

### ISSUE THREAD

**Issue by yakMM** [seddonym/import-linter#145]
Hello,

First of all, thank you for this nice tool!

I have a few use cases of circular imports protected by `if typing.TYPE_CHECKING` conditions. These imports are useful only in type checking context (when running mypy) but are flagged by import-linter.

Consider the following:

lower.py:
```py
from __future__ import annotations

import typing

if typing.TYPE_CHECKING:
    from upper import Bar

def foo(bar: Bar) -> None:
    ...
```

upper.py:
```py
from lower import foo

class Bar:
    def something(self) -> None:
        foo(self)
```

With a layer contract:
```ini
[importlinter:contract:1]
name = My two-tier layers contract
type = layers
layers=
    upper
    lower
```

The contract will fail because upper is imported from lower, despite the import call being protected behind typing.TYPE_CHECKING (it will not be evaluated at runtime.

This pattern is quite common for projects using static type-checking tools, I believe it would be nice if import-linter had an option to handle such cases.

What do you think? Would it be easy enough to implement?

**Comment by yakMM** [comment]
https://github.com/seddonym/grimp/issues/64

**Comment by seddonym** [comment]
Sorry for the delay in responding, yes you found it! I think this is a duplicate of that issue, so I'm closing.
